c++/src/kj/async-test.c++

```diff
@@ -925,33 +923,31 @@ KJ_TEST("cancel a fiber") {
   EventLoop loop;
   WaitScope waitScope(loop);
 
-#if KJ_NO_EXCEPTIONS
   // When exceptions are disabled we can't wait() on a non-void promise that throws.
   auto paf = newPromiseAndFulfiller<void>();
-#else
-  auto paf = newPromiseAndFulfiller<int>();
-#endif
 
   bool exited = false;
+  bool canceled = false;
 
   {
     Promise<StringPtr> fiber = startFiber(65536,
-        [promise = kj::mv(paf.promise), &exited](WaitScope& fiberScope) mutable {
+        [promise = kj::mv(paf.promise), &exited, &canceled](WaitScope& fiberScope) mutable {
       KJ_DEFER(exited = true);
-#if KJ_NO_EXCEPTIONS
-      promise.wait(fiberScope);
-#else
-      int i = promise.wait(fiberScope);
-      KJ_EXPECT(i == 123);
-#endif
+      try {
+        promise.wait(fiberScope);
+      } catch (kj::CanceledException) {
```

Comment:
This shouldn't be caught in the test code, it should be caught by the fiber infrastructure. In fact, I think the fiber infrastructure should require that if a fiber was canceled, the fiber exits by throwing CanceledException, not by returning normally.